napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1609 @@
+                self.data.update_opacity(i, opacity)
+            self.refresh()
+        self.events.opacity()
+
+    @property
+    def selected_shapes(self):
+        """list: list of currently selected shapes
+        """
+        return self._selected_shapes
+
+    @selected_shapes.setter
+    def selected_shapes(self, selected_shapes):
+        self._selected_shapes = selected_shapes
+        self._selected_box = self.interaction_box(selected_shapes)
+
+        # Update properties based on selected shapes
+        face_colors = list(set([self.data.shapes[i]._face_color_name for i in
+                           selected_shapes]))
+        if len(face_colors) == 1:
+            face_color = face_colors[0]
+            with self.block_update_properties():
+                self.face_color = face_color
+
+        edge_colors = list(set([self.data.shapes[i]._edge_color_name for i in
+                           selected_shapes]))
+        if len(edge_colors) == 1:
+            edge_color = edge_colors[0]
+            with self.block_update_properties():
+                self.edge_color = edge_color
+
+        edge_width = list(set([self.data.shapes[i].edge_width for i in
+                          selected_shapes]))
+        if len(edge_width) == 1:
+            edge_width = edge_width[0]
+            with self.block_update_properties():
+                self.edge_width = edge_width
+
+        opacities = list(set([self.data.shapes[i].opacity for i in
+                         selected_shapes]))
+        if len(opacities) == 1:
+            opacity = opacities[0]
+            with self.block_update_properties():
+                self.opacity = opacity
+
+    @property
+    def mode(self):
+        """MODE: Interactive mode. The normal, default mode is PAN_ZOOM, which
+        allows for normal interactivity with the canvas.
+
+        The SELECT mode allows for entire shapes to be selected, moved and
+        resized.
+
+        The DIRECT mode allows for shapes to be selected and their individual
+        vertices to be moved.
+
+        The VERTEX_INSERT and VERTEX_REMOVE modes allow for individual
+        vertices either to be added to or removed from shapes that are already
+        selected. Note that shapes cannot be selected in this mode.
+
+        The ADD_RECTANGLE, ADD_ELLIPSE, ADD_LINE, ADD_PATH, and ADD_POLYGON
+        modes all allow for their corresponding shape type to be added.
+        """
+        return self._mode
+
+    @mode.setter
+    def mode(self, mode):
+        if mode == self._mode:
+            return
+        old_mode = self._mode
+        if mode == Mode.PAN_ZOOM:
+            self.cursor = 'standard'
+            self.interactive = True
+            self.help = 'enter a selection mode to edit shape properties'
+        elif mode in [Mode.SELECT, Mode.DIRECT]:
+            self.cursor = 'pointing'
+            self.interactive = False
+            backspace = 'delete' if sys.platform == 'darwin' else 'backspace'
```

Comment:
You could add these two your constants file as well.

Reply:
done